gwcs/api.py

```diff
@@ -0,0 +1,212 @@
+        returned as rounded integers.
+        """
+        result = self.invert(*world_arrays, with_units=False)[::-1]
+        return result # astype(int)
+
+    @property
+    def array_shape(self):
+        """
+        The shape of the data that the WCS applies to as a tuple of
+        length `~BaseLowLevelWCS.pixel_n_dim`.
+        If the WCS is valid in the context of a dataset with a particular
+        shape, then this property can be used to store the shape of the
+        data. This can be used for example if implementing slicing of WCS
+        objects. This is an optional property, and it should return `None`
+        if a shape is not known or relevant.
+        The shape should be given in ``(row, column)`` order (the convention
+        for arrays in Python).
+        """
+        return self._array_shape
+
+    @array_shape.setter
+    def array_shape(self, value):
+        self._array_shape = value
+        
+    @property
+    def pixel_bounds(self):
+        """
+        The bounds (in pixel coordinates) inside which the WCS is defined,
+        as a list with `~BaseLowLevelWCS.pixel_n_dim` ``(min, max)`` tuples.
+        The bounds should be given in ``[(xmin, xmax), (ymin, ymax)]``
+        order. WCS solutions are sometimes only guaranteed to be accurate
+        within a certain range of pixel values, for example when defining a
+        WCS that includes fitted distortions. This is an optional property,
+        and it should return `None` if a shape is not known or relevant.
+        """
+        return self.bounding_box
+
+    @property
+    def axis_correlation_matrix(self):
+        """
+        Returns an (`~BaseLowLevelWCS.world_n_dim`,
+        `~BaseLowLevelWCS.pixel_n_dim`) matrix that indicates using booleans
+        whether a given world coordinate depends on a given pixel coordinate.
+        This defaults to a matrix where all elements are `True` in the absence of
+        any further information. For completely independent axes, the diagonal
+        would be `True` and all other entries `False`.
+        """
+
+        return separable.separability_matrix(self.forward_transform)
+
+    def serialized_classes(self):
+        """
+        Indicates whether Python objects are given in serialized form or as
+        actual Python objects.
+        """
+        return False
+
+    def world_axis_object_classes(self):
+        raise NotImplementedError()
+
+    def world_axis_object_components(self):
+        raise NotImplementedError()
+
+
+    # High level APE 14 API
+
+    def low_level_wcs(self):
+        """
+        Returns a reference to the underlying low-level WCS object.
+        """
+        return self
+
+    def pixel_to_world(self, *pixel_arrays):
+        """
+        Convert pixel values to world coordinates.
+        """
+        if not self.forward_transform.uses_quantity:
+            kwargs = {'with_units': True}
+        return self.__call__(*pixel_arrays, **kwargs)
```

Comment:
why the `__call__`?

Reply:
Yes, strictly speaking, it's not necessary.